sdk/ai/azure-ai-agents/azure/ai/agents/models/_patch.py

```diff
@@ -667,6 +667,82 @@ def resources(self) -> ToolResources:
         """
         return ToolResources(azure_ai_search=AzureAISearchToolResource(index_list=self.index_list))
 
+    def add_definition(
+        self,
+        index_connection_id: str,
+        index_name: str,
+        query_type: AzureAISearchQueryType = AzureAISearchQueryType.SIMPLE,
+        filter: str = "",
+        top_k: int = 5,
+        index_asset_id: Optional[str] = None,
+    ) -> None:
+        """
+        Adds a new search index configuration dynamically.
+        Raises a ValueError if an index with the same connection_id and index_name already exists.
+
+        :param index_connection_id: Index Connection ID used by tool. Allows only one connection.
+        :type index_connection_id: str
+        :param index_name: Name of Index in search resource to be used by tool.
+        :type index_name: str
+        :param query_type: Type of query in an AIIndexResource attached to this agent.
+            Default value is AzureAISearchQueryType.SIMPLE.
+        :type query_type: AzureAISearchQueryType
+        :param filter: Odata filter string for search resource.
+        :type filter: str
+        :param top_k: Number of documents to retrieve from search and present to the model.
+        :type top_k: int
+        :param index_asset_id: Index asset ID to be used by tool.
+        :type index_asset_id: Optional[str]
+        :raises ValueError: If an index with the same connection_id and index_name exists.
+        """
+        # Check if an index with the same connection_id and index_name exists
+        if any(
+            idx.index_connection_id == index_connection_id and idx.index_name == index_name for idx in self.index_list
+        ):
+            raise ValueError(
```

Comment:
Should we raise an exception in this case? Maybe we can silently return and log the warning?
```python
logger.warning(f"Index with connection_id '{index_connection_id}' and index_name '{index_name}' already exists and cannot be added again.")
```